### PR TITLE
feat(skia): (partially) implement ShapeVisual.ViewBox and fix ShapeVisual clipping

### LINKED ISSUE #18528: [Desktop/Desktop WSL] Events Not Triggered When Using Image
### Current behavior

When using Tapped, DoubleTapped, PointerPressed,... events on an `Image`, the event is not triggered on **Desktop** or **Desktop WSL**. However, the same events are working as expected on other controls such as `Border` for example.

### Expected behavior

The Tapped, DoubleTapped, PointerPressed,... events should be triggered when interacting with an `Image`, similar to how they work with other UI elements like `Border` or to how they work for all the other platforms.

### How to reproduce it (as minimally and precisely as possible)

**Sample App to Repro: [TappedEventIssue.zip](https://github.com/user-attachments/files/17467006/TappedEventIssue.zip)**

Create a new project with the following XAML and C# code:

```xml
<Grid>
  <StackPanel Spacing="10"
              HorizontalAlignment="Center"
              VerticalAlignment="Center">
    <TextBlock Text="Hello Uno Platform!"
               HorizontalAlignment="Center" />

    <Border Height="100"
            Width="100"
            Background="LightBlue"
            Tapped="Tapped_Event">
      <TextBlock Text="Tapped Test"
                 HorizontalAlignment="Center"
                 VerticalAlignment="Center" />
    </Border>

    <Image Height="100"
           Tapped="Tapped_Event"
           Source="ms-appx:///Assets/unologo.jpg" />

    <Border Height="100"
            Width="100"
            Background="LightPink"
            DoubleTapped="DoubleTapped_Event">
      <TextBlock Text="DoubleTapped Test"
                 HorizontalTextAlignment="Center"
                 TextWrapping="WrapWholeWords"
                 HorizontalAlignment="Center"
                 VerticalAlignment="Center" />
    </Border>

    <Image Height="100"
           DoubleTapped="DoubleTapped_Event"
           Source="ms-appx:///Assets/unologo.jpg" />

    <Border Height="100"
            Width="100"
            Background="LightGreen"
            PointerPressed="PointerPressed_Event">
      <TextBlock Text="PointerPressed Test"
                 HorizontalTextAlignment="Center"
                 TextWrapping="WrapWholeWords"
                 HorizontalAlignment="Center"
                 VerticalAlignment="Center" />
    </Border>

    <Image Height="100"
           PointerPressed="PointerPressed_Event"
           Source="ms-appx:///Assets/unologo.jpg" />
  </StackPanel>
</Grid>
```

```csharp
private async void Tapped_Event(object sender, Microsoft.UI.Xaml.Input.TappedRoutedEventArgs e)
{
    ContentDialog dialog = new ContentDialog
    {
        Title = "Tapped Event",
        Content = "Tapped Event succeeded!",
        PrimaryButtonText = "OK",
        XamlRoot = this.XamlRoot
    };

    await dialog.ShowAsync();
}

private async void DoubleTapped_Event(object sender, Microsoft.UI.Xaml.Input.DoubleTappedRoutedEventArgs e)
{
    ContentDialog dialog = new ContentDialog
    {
        Title = "DoubleTapped Event",
        Content = "DoubleTapped Event succeeded!",
        PrimaryButtonText = "OK",
        XamlRoot = this.XamlRoot
    };

    await dialog.ShowAsync();
}

private async void PointerPressed_Event(object sender, Microsoft.UI.Xaml.Input.PointerRoutedEventArgs e)
{
    ContentDialog dialog = new ContentDialog
    {
        Title = "PointerPressed Event",
        Content = "PointerPressed Event succeeded!",
        PrimaryButtonText = "OK",
        XamlRoot = this.XamlRoot
    };

    await dialog.ShowAsync();
}
```

### Workaround

None

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

"Uno.Sdk": "5.4.10"

### Affected platforms

Skia
- Desktop
- Desktop WSL

### IDE

Visual Studio 2022

### IDE version

Version 17.11.4

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #18528

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Composition/Composition/ContainerVisual.skia.cs [hunk 1]
```diff
@@ -9,6 +12,19 @@ public partial class ContainerVisual : Visual
...
 	private bool _hasCustomRenderOrder;
 
+	private (Rect rect, bool isAncestorClip)? _clipRectFromArrangeCycle;
+
+	/// <summary>
+	/// Layout clipping is usually applied in the element's coordinate space.
+	/// However, for Panels and ScrollViewer headers specifically, WinUI applies clipping in the parent's coordinate space.
+	/// So, isAncestorClip will be set to true for Panels and ScrollViewer headers, indicating that clipping is in parent's coordinate space.
+	/// </summary>
+	internal (Rect rect, bool isAncestorClip)? ClipRectFromArrangeCycle
```
Comment (Youssef1313) [inline src/Uno.UI.Composition/Composition/ContainerVisual.skia.cs:22]: A shorter name could be `LayoutClipRect` which also aligns with the terminology from WinUI sources.

File: src/Uno.UI/UI/Xaml/UIElement.skia.cs [hunk 2]
```diff
@@ -327,21 +327,15 @@ internal virtual void OnArrangeVisual(Rect rect, Rect? clip)
 
 			_renderTransform?.UpdateFlowDirectionTransform();
 
-			// The clipping applied by our parent due to layout constraints are pushed to the visual through the ViewBox property
-			// This allows special handling of this clipping by the compositor (cf. ShapeVisual.Render).
+			// The clipping applied by our parent due to layout constraints are pushed to the visual through the ClipRectFromArrangeCycle property
+			// This allows special handling of this clipping by the compositor (cf. ContainerVisual.Render).
 			if (clip is null)
 			{
-				visual.ViewBox = null;
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/UIElement.skia.cs:334]: ViewBox is a public API. We should make sure about the behavior on WinUI.

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/UIElement.skia.cs:334]: I've tested a bunch of cases and ViewBox seems to not be affected at all. ViewBox seems to have a very specific use case: mapping the visual to new bounds (e.g. drawing in a 200x200 rect but actually mapping the result to be in a 20x20 rectangle). That wouldn't be very compatible with the layout clipping set during the arrange cycle.

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/UIElement.skia.cs:334]: @ramezgerges Just a note, IIRC, sometimes you need to wait for the next frame after you have called `ElementCompositionPreview.GetElementVisual` to find properties initialized. I guess you might find `Size` initialized. (and from my testing it wasn't that the values are simply delayed by one frame, but really depends on when the first call to GetElementVisual was made)

**Comment by jeromelaban** [comment]
@ramezgerges this test looks odd:
https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=145111&view=ms.vss-test-web.build-test-results-tab&runId=3116491&resultId=101728&paneView=attachments

File: src/Uno.UI.Composition/Composition/SkiaHelper.skia.cs [hunk 3]
```diff
@@ -9,8 +9,8 @@ namespace Microsoft.UI.Composition
 {
 	internal static class SkiaHelper
 	{
-		private static readonly ObjectPool<SKPaint> _paintPool = new(() => new SKPaint(), 8);
-		private static readonly ObjectPool<SKPath> _pathPool = new(() => new SKPath(), 8);
+		private static readonly ObjectPool<SKPaint> _paintPool = new(() => new SKPaint());
+		private static readonly ObjectPool<SKPath> _pathPool = new(() => new SKPath());
```
[the remark this answers is not in the document]
Comment (ramezgerges) [reply src/Uno.UI.Composition/Composition/SkiaHelper.skia.cs:13]: No, but the initial `8` was a random number I thought was nice (I thought the default would be 1 or something). The default turned out to be 2 * no. of cores, which is fine too, so there's no need for an explicit size parameter.

**Comment by ramezgerges** [comment]
@jeromelaban It's just the flakiness of screenshot tests with images. I've checked locally and it's working fine.

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**